cranelift-codegen/src/isa/x86/abi.rs

```diff
@@ -648,8 +719,17 @@ fn insert_common_prologue(
             // pushed CSRs, frame pointer.
             // Also, the size of a return address, implicitly pushed by a x86 `call` instruction,
             // also should be accounted for.
+            // If any FPR are present, count them as well as necessary alignment space.
             // TODO: Check if the function body actually contains a `call` instruction.
-            let total_stack_size = (csrs.iter(GPR).len() + 1 + 1) as i64 * word_size as i64;
+            let mut total_stack_size = (csrs.iter(GPR).len() + 1 + 1) as i64 * word_size as i64;
+
+            let num_fpr = csrs.iter(FPR).len();
+            if num_fpr > 0 {
+                total_stack_size += csrs.iter(FPR).len() as i64 * types::F64X2.bytes() as i64;
+
+                // Ensure that csr stack space is 16-byte aligned for ideal SIMD value placement.
+                total_stack_size = total_stack_size & !0b1111 + 16;
+            }
 
             insert_stack_check(pos, total_stack_size, stack_limit_arg);
```

Comment:
Unrelated, but because I noticed: do we know of anyone using `StackLimit`? `total_stack_size` here undercounts the real amount of space used in this function, as it doesn't include locals.

Reply:
All usages of `ArgumentPurpose::StackLimit` on github are either from Cranelift itself, or a gecko vendoring of Cranelift.